modules/knock_down/action.sqf

```diff
@@ -0,0 +1,52 @@
+// AUTHOR: StatusRed (EM-Creations.co.uk)
+
+private _validWeapons = (_this select 0);
+private _reviveAction = (_this select 1);
+
+private ["_nearestunits", "_nearestunitofside", "_prevAnim"];
+
+_prevAnim = animationState player; // Get the player's current animation
+
+_nearestUnits = nearestObjects [player, ["Man"], 3]; // Get all units within range
+
+_useUnits = []; // Instanatiate an array for units which can be targetted
+
+// Check that they have a suitable weapon to knock down with
+if ((primaryWeapon player) in _validWeapons) then { // If the player has a valid weapon
+	if(count _nearestUnits > 0) then { // If there's at least one man nearby
+		{
+			_unit = _x;
+			//_isOut = _unit getVariable "knockedDown";
+			_isOut = false; // The above isn't working at the moment
+			if (((side player) != (side _unit)) && !_isOut) then { // If the unit is not the player, not on the same side and not already hit
```

Comment:
```suggestion
			if (!_isOut && {(side player) != (side _unit)}) then { // If the unit is not the player, not on the same side and not already hit
```

Lazy evaluation, less intensive condition first (although this is a WIP condition)

Reply:
That's interesting, so does SQF not short circuit expressions, like other languages do?

Reply:
No, it chooses A, then B, then C, etc.
code inside {} for additional conditions is parsed but not executed, so you can have more intensive commands but they need to be in order of intensiveness, if possible.

Reply:
Good to know thanks.

Reply:
Done.